Steam version on older linux distributions requires Proton 7.x compatibility mode

Bitburner won't start on debian11 linux as of the 2023-02-22 update in steam.
To try and debug things I did try starting from the terminal mimicking what steam did.

See stacktrace from this terminal run below:
```
Uncaught Exception:
Error: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /home/karl/.steam/debian-installation/steamapps/common/Bitburner/resources/app/lib/greenworks-linux64.node)
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:2:1822)
    at Module._extensions..node (node:internal/modules/cjs/loader:1226:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:2:1822)
    at Module.load (node:internal/modules/cjs/loader:1011:32)
    at Module._load (node:internal/modules/cjs/loader:846:12)
    at f._load (node:electron/js2c/asar_bundle:2:13328)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/karl/.steam/debian-installation/steamapps/common/Bitburner/resources/app/greenworks.js:17:43)
    at Module._compile (node:internal/modules/cjs/loader:1141:14)
[3647904:0222/080429.183504:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
```

What do I need to do to run this? Downloaded and tried `./bitburner` and got:

```
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module 'electron-log'
Require stack:
- /home/erk24/Downloads/bitburner/resources/app/main.js
- 
    at Module._resolveFilename (node:internal/modules/cjs/loader:963:15)
    at n._resolveFilename (node:electron/js2c/browser_init:2:109411)
    at Module._load (node:internal/modules/cjs/loader:811:27)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/erk24/Downloads/bitburner/resources/app/main.js:4:13)
    at Module._compile (node:internal/modules/cjs/loader:1141:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
    at Module.load (node:internal/modules/cjs/loader:1011:32)
[277566:0303/075818.294401:ERROR:ozone_platform_x11.cc(239)] Missing X server or $DISPLAY
[277566:0303/075818.294454:ERROR:env.cc(255)] The platform failed to initialize.  Exiting.
The futex facility returned an unexpected error code.
Aborted (core dumped)
```

I tested by replacing my .steam/debian-installation/steamapps/common/Bitburner directory with the files in the artifact and got the same error as kodingkoning:

`A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module 'electron-log'
Require stack:
- /home/robertom/.steam/debian-installation/steamapps/common/Bitburner/resources/app/main.js
- 
    at Module._resolveFilename (node:internal/modules/cjs/loader:963:15)
    at n._resolveFilename (node:electron/js2c/browser_init:2:109411)
    at Module._load (node:internal/modules/cjs/loader:811:27)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/robertom/.steam/debian-installation/steamapps/common/Bitburner/resources/app/main.js:4:13)
    at Module._compile (node:internal/modules/cjs/loader:1141:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
    at Module.load (node:internal/modules/cjs/loader:1011:32)
libva error: /usr/lib/x86_64-linux-gnu/dri/nvidia_drv_video.so has no function __vaDriverInit_1_0
[11451:0307/100131.345710:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
[11420:0307/100136.451808:ERROR:nss_util.cc(349)] After loading Root Certs, loaded==false: NSS error code: -8018`

If I copy just the bitburner executable into my original directory, I get the original `GLIBCXX_3.4.29' not found` error.

The original bitburner file and the one from the artifact are the same size and diff reports no differences.

FYI, the game does play for me if I use the Proton 7.x compatibility mode.

--- a/electron/main.js
+++ b/electron/main.js
@@ -28,7 +28,7 @@ const debounce = require("lodash/debounce");
 const Store = require("electron-store");
 const store = new Store();
 const path = require("path");
-const { realpathSync } = require("fs");
+const { realpathSync, readFileSync } = require("fs");
 const { fileURLToPath } = require("url");
 
 log.transports.file.level = store.get("file-log-level", "info");
@@ -188,8 +188,8 @@ app.on("ready", async () => {
     let relativePath;
     /**
      * "realpathSync" will throw an error if "filePath" points to a non-existent file. If an error is thrown here, the
-     * electron app will crash immediately. We can use fs.existsSync to check "filePath" before using it, but it's best
-     * to try-catch the entire code block and avoid unexpected issues.
+     * Electron app will not write any error logs, and the request will fail silently. We can use fs.existsSync to check
+     * "filePath" before using it, but it's best to try-catch the entire code block.
      */
     try {
       filePath = fileURLToPath(url);
@@ -197,13 +197,7 @@ app.on("ready", async () => {
       relativePath = path.relative(__dirname, realPath);
       // Only allow access to files in "dist" folder or html files in the same directory
       if (method === "GET" && (relativePath.startsWith("dist") || relativePath.match(/^[a-zA-Z-_]*\.html/))) {
-        /**
-         * By default, requests made by net.fetch go through custom protocol handlers, so we have to explicitly tell it
-         * to bypass those handles; otherwise, it creates an infinite loop.
-         *
-         * Ref: https://github.com/electron/electron/issues/39402
-         */
-        return net.fetch(realPath, { bypassCustomProtocolHandlers: true });
+        return new Response(readFileSync(realPath));
       }
     } catch (error) {
       log.error(error);

--- a/electron/package.json
+++ b/electron/package.json
@@ -24,7 +24,7 @@
     "buildResources": "public"
   },
   "dependencies": {
-    "@catloversg/steamworks.js": "0.0.1",
+    "@catloversg/steamworks.js": "0.0.2",
     "electron-log": "^4.4.8",
     "electron-store": "^8.1.0",
     "lodash": "^4.17.21"

